File: FastMines_core/src/fmg/core/img/IAnimatedController.cs
using System;

namespace Fmg.Core.Img {

    /// <summary> Image MVC: animate controller </summary>
    /// <typeparam name="TImage">platform specific view/image/picture or other display context/canvas/window/panel</typeparam>
    /// <typeparam name="TImageView">image view</typeparam>
    /// <typeparam name="TImageModel">image model</typeparam>
    public interface IAnimatedController<out TImage, out TImageView, out TImageModel>
                      : IImageController<    TImage,     TImageView,     TImageModel>
        where TImage      : class
        where TImageView  : IImageView<TImage, TImageModel>
        where TImageModel : IAnimatedModel
    {

        void AddModelTransformer(IModelTransformer transformer);
        void RemoveModelTransformer(Type /** extends IModelTransformer */ transformerClass);

        void UseRotateTransforming(bool enable);

        void UsePolarLightFgTransforming(bool enable);

    }

}

File: FastMines_core/src/fmg/core/img/IAnimatedController.java
package fmg.core.img;

/**
 * Image MVC: animate controller
 *
 * @param <TImage> platform specific view/image/picture or other display context/canvas/window/panel
 * @param <TImageView> image view
 * @param <TImageModel> image model
 */
public interface IAnimatedController<TImage,
                                     TImageView  extends IImageView<TImage, TImageModel>,
                                     TImageModel extends IAnimatedModel>
                 extends IImageController<TImage, TImageView, TImageModel>
{

    void addModelTransformer(IModelTransformer transformer);
    void removeModelTransformer(Class<? extends IModelTransformer> transformerClass);

    void useRotateTransforming(boolean enable);
    void usePolarLightFgTransforming(boolean enable);

}
